doc/developers/contributing.rst

```diff
@@ -428,19 +428,46 @@ Documentation
 
 We are glad to accept any sort of documentation: function docstrings,
 reStructuredText documents (like this one), tutorials, etc. reStructuredText
-documents live in the source code repository under the doc/ directory.
-
+documents live in the source code repository under the ``doc/`` directory.
 You can edit the documentation using any text editor, and then generate the
-HTML output by typing ``make html`` from the doc/ directory. Alternatively,
-``make html-noplot`` can be used to quickly generate the documentation without
-the example gallery. The resulting HTML files will be placed in _build/html/
-and are viewable in a web browser. See the README file in the doc/ directory
-for more information.
-
-For building the documentation, you will need `sphinx
-<http://www.sphinx-doc.org/en/stable/>`_,
-`matplotlib <http://matplotlib.org>`_ and
-`pillow <http://pillow.readthedocs.io/en/latest/>`_.
+HTML output by building the documentation website.
+
+**Building the documentation**
+
+Building the documentation requires the ``sphinx``, ``sphinx-gallery``,
+``numpydoc``, and ``matplotlib`` packages::
+
+    pip install sphinx numpydoc sphinx-gallery matplotlib
+
+It also requires having the version of scikit-learn installed that corresponds
+to the documentation, e.g.::
+
+    pip install --editable ..
+
+To generate the full web site, including the example gallery (this might take a
+while)::
+
+    make html
+
+Or, if you'd rather quickly generate the documentation without the example
+gallery::
+
+    make html-noplot
+
+That should create all the documentation in the ``_build/html/stable`` directory.
+
+To build the PDF manual, run::
+
+    make latexpdf
+
+**Hosting and automatic builds**
+
+The website is hosted at github, but should rarely be updated manually
+by pushing to the https://github.com/scikit-learn/scikit-learn.github.io repository.
+
+Most updates can be made by pushing to master (for /dev) or a release branch
+like 0.99.X, from which Circle CI builds and uploads documentation. (See the
+Developer Documentation for further details.)
```

Comment:
Do you think this section should be moved back to `scikit-learn/doc/README`? Or is there a more appropriate place to put it?

Reply:
Haha phone did a strange thing to my words.

No this doesn't belong in readme. It's for maintainers. If it's not covered in maintainers.rst it can go there, otherwise here is fine, but it should not refer to developer docs

Reply:
Hah I was wondering about that. 

There's a reference to this in `maintainer.rst`, but it's pretty sparse. I'll move this section over to `maintainer.rst`.